criu/files.c

```diff
@@ -582,6 +585,11 @@ static int dump_one_file(struct pid *pid, int fd, int lfd, struct fd_opts *opts,
 		return do_dump_gen_file(&p, lfd, ops, e);
 	}
 
+	if (p.fs_type == PID_FS_MAGIC) {
```

Comment:
You need to move this piece of code a bit higher, because your pidfds (on 6.9+ kernels) do not reach that point. Instead, they go to `dump_unsupp_fd(&p, lfd, "reg", link.name + 1, e);` and fail later on. Just move this to the line 555, right after `if (p.fs_type == ANON_INODE_FS_MAGIC) {` block ends.

Reply:
see also https://github.com/torvalds/linux/commit/db3d841ac9edb0b98cc002e3b27c0b266ecfe5ba

Reply:
Done!